src/middlewares/schema.spec.ts

```diff
@@ -200,7 +233,57 @@ describe('on schema that contains fields with bail methods', () => {
       },
     });
 
-    const { context } = await schema[0].run({ params: {} });
+    const { context } = await schema[0].run({ query: {} });
+    expect(context.errors).toHaveLength(1);
+  });
+
+  it('stops validation chain if any validation fails with "exist" validator', async () => {
+    const allLocations = validLocations.length;
+    const schema = checkSchema({
+      foo: {
+        exists: {
+          bail: true,
+          errorMessage: 'exists checking failed',
+        },
+        isLength: {
+          options: {
+            min: 5,
+          },
+        },
+      },
+    });
+
+    const { context } = await schema[0].run({ query: {} });
+    // when location is not specified, checking each location one by one,
+    // if the expected field does not show up at all in any location, then there will be 5 errors.
+    // finally bails.
+    expect(context.errors).toHaveLength(allLocations);
```

Comment:
nit: we can hardcode `allLocations` in the tests since locations will unlikely change in length.

Reply:
sure 👌